### ISSUE TITLE
Upgrade Cucumber version to 4

### ISSUE THREAD

**Issue by borbamartin** [serenity-bdd/serenity-cucumber#178]
Hey there!
Is there any chance the Cucumber JVM version could be upgraded to the latest? I'd like to be able to use `cucumber.api.TypeRegistryConfigurer`.

Thanks!

**Comment by scotty6435** [comment]
+1 for this in general. Cucumber 4 supports parallel execution which is massive especially as the cucumber-jvm plugin is pretty janky

**Comment by wakaleo** [comment]
We are well aware of the parallel execution, but breaking thousands of existing features is a show-stopper, and that needs a lot of work to get around.

**Comment by scotty6435** [comment]
@wakaleo, completely understandable. Target for next major release?

**Comment by wakaleo** [comment]
Yes, it is targeted for 2.1. The code is mostly in a branch, but it needs a lot of testing to ensure that it can work for both Cucumber 4.x and Cucumber 2.x with the same serenity-core version.

**Comment by mlvandijk** [comment]
Note that the `TypeRegistryConfigurer` was introduced in Cucumber-jvm v3.0.0, see [CHANGELOG.md](https://github.com/cucumber/cucumber-jvm/blob/master/CHANGELOG.md)

**Comment by mleegwt** [comment]
#175 There is a pull request to update to v4. Acording to https://docs.cucumber.io/cucumber/step-definitions/#expressions the regexp style step definitions should still be possible. What are the incompatibilities? Maybe we can solve them together.

**Comment by cliviu** [comment]
> #175 There is a pull request to update to v4. Acording to https://docs.cucumber.io/cucumber/step-definitions/#expressions the regexp style step definitions should still be possible. What are the incompatibilities? Maybe we can solve them together.

Hi, here you can find the incompatibility.

**Comment by mleegwt** [comment]
According to https://github.com/cucumber/website/blob/master/apps/dynamic/views/_posts/announcing-cucumber-jvm-4-2-0.md de backwardscompatibility is reintroduced.

**Comment by wakaleo** [comment]
We are looking into this - there is still work you need to do to upgrade, so it will still need a separate module.

**Comment by KLBonn** [comment]
Hi folks,
we currently are upgrading to Cucumber4, and would like to continue using the Serenity Jira two-way integration.
I understand that this still is "work in progress", however I wanted to try out whatever is done yet.
So I cloned https://github.com/serenity-bdd/serenity-cucumber4, did a build/install (to my local Maven repo), and executed my automation project.

And I am happy to say: "Works on my machine!" :-)

So, can you shed some light on the tasks/issues holding you from publishing a release to Maven Central?

**Comment by wakaleo** [comment]
Good to hear - the delay is mainly that I haven’t had time to set up the build pipeline for this module yet.

**Comment by turturie** [comment]
Hi guys,
Any update?
@wakaleo , I did follow the steps you mentioned here,
https://johnfergusonsmart.com/running-parallel-tests-serenity-bdd-cucumber/,
but no luck, I am having issues with Scenario Outline not been reported correctly.

If there anything I can do to help to implement Cucumber native threads, let me know!

Thanks for this great FW.
